### PR TITLE
[R20-1321] Add custom collection

### PR DESCRIPTION
**Pull request by dylankelly**
**Issue**: https://digital-vic.atlassian.net/browse/R20-1321

### What I did

- Added TideCustomCollection component - this will be used for custom search experiences in Landing page content types. It will be used first for the Cameras save lives location table.
- Adds API endpoint for proxying elasticsearch requests seperate to app search (would be good to also consolidate content collection in a follow up PR)
- Refactors search page components to be consistent

### How to test

- Added integration tests
- To test manually
- - Set `NUXT_PUBLIC_TIDE_ELASTICSEARCH_HOST=https://a83890f7a31dea14e1ae83c6f0afacca.sdp2-b.elastic.sdp.vic.gov.au`
- - Create landing page with Custom collection component - See examples/nuxt-app/test/fixtures/landingpage/custom-collection.json props object for JSON config

#### For all PR's

- [ ] I've added relevant changes to the project Readme if needed.
- [ ] I've updated the documentation site as needed.
- [ ] I have added unit tests to cover my changes (if not applicable, please state why in a comment)

#### For new components only

- [ ] I have added a story covering all variants
- [ ] I have checked a11y tab in storybook passes
- [ ] Any events are emitted on the event bus

### CODE REVIEW

File: examples/nuxt-app/test/fixtures/landingpage/custom-collection.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      "id": "123",
+      "title": "Cameras save lives",
+      "props": {
+        "link": {
+          "url": "https://www.vic.gov.au",
+          "text": "View more"
+        },
+        "searchListingConfig": {
+          "searchProvider": "elasticsearch",
+          "index": "sdp_data_pipelines_scl",
```
Comment (dylankelly) [inline examples/nuxt-app/test/fixtures/landingpage/custom-collection.json:111]: add the node index as default

**Review by waitingallday — APPROVED** [review]

**Comment by lambry** [comment]
Hey @dylankelly, 

- Should the autocomplete be working? (and debounced)
- Might needs a little spacing here Screenshot 2023-08-28 at 10 42 32 am

Cheers

**Comment by dylankelly** [comment]
> Hey @dylankelly,
> 
> * Should the autocomplete be working? (and debounced)
> * Might needs a little spacing here Screenshot 2023-08-28 at 10 42 32 am
> 
> Cheers

Autocomplete won't be a thing. I've advised Jess. Have fixed up spacing

**Review by jeffdowdle — DISMISSED** [review]
I couldn't get search listing running locally, just setting this to 'request changes' so that this doesn't get merged with possible issues

**Comment by dylankelly** [comment]
> I couldn't get search listing running locally, just setting this to 'request changes' so that this doesn't get merged with possible issues

@jeffdowdle this had an issue with the proxy setup - fixed [here](https://github.com/dpc-sdp/ripple-framework/pull/777/commits/2c0da907acf1f2d2b98955344e3fbde321f185bd) 

Have put test .env in slack

**Review by lambry — APPROVED** [review]

**Pull request merged**